api/dbadapters/bigquery.ts

```diff
@@ -190,6 +195,28 @@ export class BigQueryDbAdapter implements IDbAdapter {
 
   public async close() {}
 
+  public async persistedStateMetadata(
+    compiledGraph: dataform.ICompiledGraph
+  ): Promise<dataform.IPersistedTableMetadata[]> {
+    const adapter = adapters.create(
+      compiledGraph.projectConfig,
+      compiledGraph.dataformCoreVersion || "1.0.0"
+    );
+    try {
+      const { rows } = await this.runQuery(
+        `SELECT * FROM \`${adapter.resolveTarget({ name: STATE_PERSIST_TABLE_NAME })}\``,
```

Comment:
I thought the intention is this table is fixed regardless of environments etc. The problem here is, this will create a different table for each environment (schemaSuffix) and in the future will also end up respecting tablePrefix option we add.

The design doc implies that the schema is fixed regardless of the environment, in which case you should have a fixed schema name and table name and pass both in to resolveTarget.

My suggestion would be to create a table:
schema: dataform_meta
table: cache_state

Can we add this to the design doc?

Reply:
Added: https://docs.google.com/document/d/1cksnd-lFnQeWLIbhEaYAENl3ylKWcmsm-JMeUXIgabU/edit?ts=5e3d3f19#heading=h.2g1udzmih3xp